src/SprykerEco/Zed/PunchoutCatalogs/Persistence/PunchoutCatalogsEntityManager.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Persistence;
+
+use Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer;
+use Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection;
+use Spryker\Zed\Kernel\Persistence\AbstractEntityManager;
+
+/**
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsPersistenceFactory getFactory()
+ */
+class PunchoutCatalogsEntityManager extends AbstractEntityManager implements PunchoutCatalogsEntityManagerInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer
+     */
+    public function createPunchoutCatalogConnection(PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer): PunchoutCatalogConnectionTransfer
+    {
+        $punchoutCatalogConnectionEntity = $this->getFactory()
+            ->createPunchoutCatalogsConnectionMapper()
+            ->mapPunchoutCatalogConnectionTransferToEntity(
+                $punchoutCatalogConnectionTransfer,
+                new PgwPunchoutCatalogConnection()
+            );
+
+        $punchoutCatalogConnectionEntity->save();
+
+        return $this->getFactory()
+            ->createPunchoutCatalogsConnectionMapper()
+            ->mapPunchoutCatalogConnectionEntityToTransfer($punchoutCatalogConnectionEntity, $punchoutCatalogConnectionTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer
+     *
+     * @return bool
+     */
+    public function updatePunchoutCatalogConnection(PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer): bool
```

Comment:
Don't you think, that in this case, `updatePunchoutCatalogConnection()` method should expect a valid PunchoutCatalogConnectionTransfer.

The responsibility of EntityManager are operations on entities. Here, you're mixing READ/CHECK behavior with UPDATE behavior. I would suggest moving the entity existence check to the business layer, and call EntityManager only when your entity is valid.

Please argue, if you disagree.

Reply:
I agree with your concept but it will lead to 2 DB requests instead of one.